src/runtime/runtime-debug.cc

```diff
@@ -1090,57 +1102,42 @@ RUNTIME_FUNCTION(Runtime_RecordReplayAssertExecutionProgress) {
     RecordReplayOnTargetProgressReached();
   }
 
-  if (!gRecordReplayAssertProgress) {
+  if (!gRecordReplayCheckProgress) {
```

Comment:
This function is written assuming that gRecordReplayAssertProgress implies gRecordReplayCheckProgress, but per the earlier comment we don't want to make that assumption, and should treat these as two independent flags/counters.

Reply:
I guess having 2 branches instead of 1 in this function is not going to make too bad of a performance impact?